application/apps/indexer/addons/file-tools/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+use anyhow::Result;
+use std::{
+    fs::{metadata, File},
+    io::Read,
+    path::Path,
+    str::from_utf8,
+};
+
+const BYTES_TO_READ: u64 = 10240;
+
+pub fn is_binary(file_path: String) -> Result<bool> {
+    let chunks = fetch_starting_chunk(Path::new(&file_path));
+    let buffer = match chunks {
+        Ok(buffer) => buffer,
+        Err(err) => return Err(err),
+    };
+
+    let result = from_utf8(&buffer);
+    match result {
+        Ok(_file_content) => Ok(false),
+        Err(_err) => Ok(true),
+    }
+}
+
+fn fetch_starting_chunk(file_path: &Path) -> Result<Vec<u8>> {
+    let file = File::open(file_path)?;
+    let file_length: u64 = metadata(file_path)?.len() - 1;
+    let file_length = if BYTES_TO_READ < file_length {
+        BYTES_TO_READ
+    } else {
+        file_length
+    };
+
+    let mut file = file.take(file_length);
+    let mut buffer = Vec::new();
+    file.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
```

Comment:
Please add tests for error/specific use cases:
- the wrong path is given (file doesn't exist)
- file has a size 0 bytes